lib/tbd/psi.rb

```diff
@@ -144,113 +152,256 @@ def initialize
+        joint:        0.140000, # Detail 7.4.2
+        transition:   0.000000  # defaults to 0
       }.freeze
 
-      # NECB-QC (CCQ, Chapitre i1 ... "Ci1"), code-compliant defaults.
+      # CCQ, Chapitre I1, code-compliant defaults.
       @set["code (Quebec)"] =
       {
-        rimjoist:      0.300, # re Ci1
-        parapet:       0.325, # re Ci1
-        fenestration:  0.200, # re Ci1
-        corner:        0.300, # inferred from description, not explicitely set
-        balcony:       0.500, # re Ci1
-        party:         0.450, # re Ci1
-        grade:         0.450, # re Ci1
-        joint:         0.200, # re Ci1
-        transition:    0.000  # defaults to 0
+        rimjoist:     0.300000, # re I1
+        parapet:      0.325000, # re I1
+        roof:         0.325000, # same as parapet
+        fenestration: 0.200000, # re I1
+        corner:       0.300000, # inferred from description, not explicitely set
+        balcony:      0.500000, # re I1
+        balconysill:  0.500000, # inferred, same as balcony
+        party:        0.450000, # re I1
+        grade:        0.450000, # re I1
+        joint:        0.200000, # re I1
+        transition:   0.000000  # defaults to 0
       }.freeze
 
-      # NECB-QC (CCQ, Chapitre i1), non-code-compliant defaults.
+      # CCQ, Chapitre I1, non-code-compliant defaults.
       @set["uncompliant (Quebec)"] =
       {
-        rimjoist:      0.850, # re Ci1
-        parapet:       0.800, # re Ci1
-        fenestration:  0.500, # re Ci1
-        corner:        0.850, # inferred from description, not explicitely set
-        balcony:       1.000, # re Ci1
-        party:         0.850, # re Ci1
-        grade:         0.850, # re Ci1
-        joint:         0.500, # re Ci1
-        transition:    0.000  # defaults to 0
+        rimjoist:     0.850000, # re I1
+        parapet:      0.800000, # re I1
+        roof:         0.800000, # same as parapet
+        fenestration: 0.500000, # re I1
+        corner:       0.850000, # inferred from description, not explicitely set
+        balcony:      1.000000, # re I1
+        balconysill:  1.000000, # inferred, same as balcony
+        party:        0.850001, # re I1
+        grade:        0.850001, # re I1
+        joint:        0.500001, # re I1
+        transition:   0.000000  # defaults to 0
+      }.freeze
+
+      # ASHRAE 90.1 2022 (A10) "default" steel-framed and metal buildings.
+      @set["90.1.22|steel.m|default"] =
+      {
+        rimjoist:     0.307000, # "intermediate floor to wall intersection"
+        parapet:      0.260000, # "parapet" edge
+        roof:         0.020000, # (non-parapet) "roof edge"
+        fenestration: 0.194000, # "wall to vertical fenestration intersection"
+        corner:       0.000000, # (unspecified, defaults to 0)
+        balcony:      0.307001, # "intermediate floor balcony/overhang" edge
+        balconysill:  0.307002, # "intermediate floor balcony" edge (when sill)
```

Comment:
Adds ASHRAE 90.1 2022 (A10) _default_ vs _unmitigated_ PSI factor sets.

The addition of "1" or "2" at the 6th decimal place is proposed as a means to prioritize edge types, when 2 or more types have the same PSI factor. For instance, the following edge types in ASHRAE 90.1 2022 A10 (_default_ for "steel-framed and metal buildings"):

- intermediate floor to wall intersections
- intermediate floor balconies overhangs
- intermediate floor balconies + _vertical fenestration_

... all have assigned PSI factors of 0.307 W/K per meter. TBD keeps track of all possible types that could apply to a given edge, yet ultimately selects the type based on the highest applicable PSI factor. The determination is somewhat random when 2 or more types have equal PSI factors. By slightly incrementing (or decrementing) the PSI factor this way, the selection is deterministic. Incrementing a PSI factor by 1/100,000th has a negligible impact on thermal bridging loss allocation.